iib/workers/tasks/opm_operations.py

```diff
@@ -907,3 +911,238 @@ def verify_operator_exists(
+
+
+def deprecate_bundles(
+    bundles: List[str],
+    base_dir: str,
+    binary_image: str,
+    from_index: str,
+    overwrite_target_index_token: Optional[str] = None,
+    container_tool: Optional[str] = None,
+) -> None:
+    """
+    Deprecate the specified bundles from the index image.
+
+    Only Dockerfile is created, no build is performed.
+
+    :param list bundles: pull specifications of bundles to deprecate.
+    :param str base_dir: base directory where operation files will be located.
+    :param str binary_image: binary image to be used by the new index image.
+    :param str from_index: index image, from which the bundles will be deprecated.
+    :param str overwrite_target_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required to use ``overwrite_target_index``.
+        The format of the token must be in the format "user:password".
+    :param str container_tool: the container tool to be used to operate on the index image
+    """
+    from iib.workers.tasks.utils import run_cmd, set_registry_token
+
+    cmd = [
+        opm_path,
+        'index',
+        'deprecatetruncate',
+        '--generate',
+        '--binary-image',
+        binary_image,
+        '--from-index',
+        from_index,
+        '--bundles',
+        ','.join(bundles),
+        '--allow-package-removal',
+    ]
+    if container_tool:
+        cmd.append('--container-tool')
+        cmd.append(container_tool)
+    with set_registry_token(overwrite_target_index_token, from_index):
+        run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to deprecate the bundles')
+
+
+def _find_index_version(index_image: str) -> str:
+    from iib.workers.tasks.utils import get_image_label
+    index_image_version = index_image.split(":")[1]
+    if index_image_version.startswith("v4."):
+        return index_image_version
+    else:
+        index_version = get_image_label(index_image, 'com.redhat.index.delivery.version')
+        return index_version
+
+
+def set_opm_verison(from_index: Optional[str] = None):
+    """
+    This function will verify if opm config is set in worker
+    if the config entry exists for the index image version
+    and return the opm version path to be used in the entire IIB request.
+    :param str from_index: from_index_image for the request
+
+    """
+    global opm_path
+    log.info("Determining the OPM version to use")
+    worker_config = get_worker_config()
+    opm_paths_config = worker_config.get("iib_ocp_opm_mapping")
+    if opm_paths_config is None or from_index is None:
+        log.warning("Either iib_ocp_opm_mapping config is not set or from_index is not set, using the default opm")
+        return
+    index_version =_find_index_version(from_index)
+    opm_path = opm_paths_config.get(index_version, opm_paths_config["default"])
+    log.info("opm_path set to %s", opm_path)
+
+
+def set_opm(func):
```

Comment:
docstring

And I guess this is just for testing. There are too many `.warning()` for debug.

Reply:
Yeah I missed removing those before committing, updated the patch